### PR TITLE
Automate/improve build, releases, and testing.

### PR DESCRIPTION
**Pull request by zachferland**

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    frameworks: [ 'mocha', 'chai' ],
+    files: [
+      'test/uportsubprovider.js'
+      // Current tooling and testing approach does not allow all tests to be run
+      // in broswer enviroments.
+      // 'test/*.js'
+    ],
+    preprocessors: {
+        'test/uportsubprovider.js': [ 'webpack', 'sourcemap']
+        // 'test/*.js': [ 'webpack', 'sourcemap' ]
```
Comment (oed) [inline karma.conf.js]: Why are we only runing the uportsubprovider test?

**Comment by oed** [comment]
Is there a need to run the tests in node at all? This library is only intended to be used in the browser anyway right?
We talked breifly about moching a http server for the msgServer.js tests. Is that something you think should be in this PR?

File: karma.conf.js (hunk 1 above)
Comment (zachferland) [inline karma.conf.js]: in karma (web env) yes, because the test files having dependencies which are node specific, we should look to getting all those tests running in a web env (karma) as soon as possible. Can be another PR related to testing.

Comment (oed) [inline karma.conf.js]: Hm, but the component works in a web env right? Otherwise the library would be unusable in webapps.

**Comment by zachferland** [comment]
currently/originally the tests were only run in node, the goal would be run them all in both node/browser env. Not sure if only intended for only browser use, its only case that makes sense at this point in time, but we should have an extended discussion considering what it will look like in the future and if there is possible reasons to run in a node env. 

Yeah we definitely mock those requests because right now the tests are slow and unpredictable. I think that can be in separate testing PR, also included with the improvements mentioned above. 

Also note, this PR does not actually activate automated deployments, so we don't have worry yet about untested code be unexpectedly released.

File: karma.conf.js (hunk 1 above)
Comment (zachferland) [inline karma.conf.js]: yup the components work in browser env, but when running the tests you also have build/bundle the tests files, and it is the tests files specifically which rely on dependencies that don't allow us to run them in browser
